cdap-client/src/main/java/co/cask/cdap/client/ScheduleClient.java

```diff
@@ -70,6 +74,49 @@ public ScheduleClient(ClientConfig config) {
   }
 
   @Deprecated
+  public void add(Id.Schedule schedule, ScheduleInstanceConfiguration configuration) throws IOException,
+    UnauthenticatedException, NotFoundException, UnauthorizedException, AlreadyExistsException {
+    add(schedule.toEntityId(), configuration);
+  }
+
+  public void add(ScheduleId scheduleId, ScheduleInstanceConfiguration configuration) throws IOException,
+    UnauthenticatedException, NotFoundException, UnauthorizedException, AlreadyExistsException {
+
+    String path = String.format("apps/%s/versions/%s/schedules/%s",
+                                scheduleId.getApplication(), scheduleId.getVersion(), scheduleId.getSchedule());
+    URL url = config.resolveNamespacedURLV3(scheduleId.getNamespaceId(), path);
+    HttpRequest request = HttpRequest.put(url).withBody(GSON.toJson(configuration)).build();
+    HttpResponse response = restClient.execute(request, config.getAccessToken(),
+                                               HttpURLConnection.HTTP_NOT_FOUND,
+                                               HttpURLConnection.HTTP_CONFLICT);
+    if (HttpURLConnection.HTTP_NOT_FOUND == response.getResponseCode()) {
+      throw new NotFoundException(scheduleId);
+    } else if (HttpURLConnection.HTTP_CONFLICT == response.getResponseCode()) {
+      throw new AlreadyExistsException(scheduleId);
+    }
+  }
+
+  @Deprecated
+  public void update(Id.Schedule schedule, ScheduleInstanceConfiguration configuration) throws IOException,
+    UnauthenticatedException, NotFoundException, UnauthorizedException, AlreadyExistsException {
+    add(schedule, configuration);
+  }
+
+  public void update(ScheduleId scheduleId, ScheduleInstanceConfiguration configuration) throws IOException,
+    UnauthenticatedException, NotFoundException, UnauthorizedException, AlreadyExistsException {
+
+    String path = String.format("apps/%s/versions/%s/schedules/%s/update",
+                                scheduleId.getApplication(), scheduleId.getVersion(), scheduleId.getSchedule());
+    URL url = config.resolveNamespacedURLV3(scheduleId.getNamespaceId(), path);
+    HttpRequest request = HttpRequest.post(url).withBody(GSON.toJson(configuration)).build();
+    HttpResponse response = restClient.execute(request, config.getAccessToken(),
+                                               HttpURLConnection.HTTP_NOT_FOUND);
```

Comment:
does it ever return bad request? For example if you update a time schedule to be a stream schedule?